### ISSUE TITLE
int counter logging shouldn't be reduced across ranks 

### ISSUE THREAD

**Issue by stas00** [NVIDIA/NeMo#7665]
**Describe the bug**

It appears that Nemo is using the wrong PTL API for logging aggregate int counters:

```
$ grep '\.log(' | grep global 
nemo/collections/nlp/models/language_modeling/megatron_finetune_model.py:            
self.log('global_step', self.trainer.global_step, prog_bar=True, rank_zero_only=True, batch_size=1)
[...]
$ grep '\.log(' | grep global | grep -v float | wc -l
10
```
same happens to other counters like `consumed_samples`

Currently PTL emits a warning:

```
site-packages/pytorch_lightning/trainer/connectors/logger_connector/result.py:212: UserWarning: You 
called `self.log('consumed_samples', ...)` in your `training_step` but the value needs to be floating 
point. Converting it to torch.float32.
pytorch_lightning/trainer/connectors/logger_connector/result.py:212: UserWarning: You called 
`self.log('global_step', ...)` in your `training_step` but the value needs to be floating point. 
Converting it to torch.float32.
```

Leading to both - an incorrect data type in the log and a wasteful reduction across ranks, even though all ranks should have the same counter value.

I first asked about this at PTL and their detailed answer on how to do it correctly is here:
https://github.com/Lightning-AI/lightning/issues/18739#issuecomment-1753263694

tldr: for non-reduced counters the call should be: `trainer.logger.log_metrics()` and not `self.log()` 

Thank you.

**Comment by nithinraok** [comment]
@athitten FYI
